QuizApp/Data/DataSources/SecureStorageProtocol.swift

```diff
@@ -14,4 +14,8 @@ protocol SecureStorageProtocol {
 
     func set(_ value: Data, for key: String)
 
+    func delete(_ key: String)
+
+    func clear()
```

Comment:
I would advise against exposing a clear function because you might accidentally delete something you do not want.
If you are setting via key, why can't you delete via key?